tests/test_core.py

```diff
@@ -627,10 +627,54 @@ def test_interpolate(test_pd_df):
     assert all([True if isstr(i) else ~np.isnan(i) for i in df.data.foo])
 
 
+def test_interpolate_time_exists(test_pd_df):
+    _df = test_pd_df.copy()
+    df = IamDataFrame(_df)
```

Comment:
You can use `test_df_year` to avoid having to cast to an IamDataFrame in the test... (Also, `test_df_pdf` already serves a copy).

```suggestion
def test_interpolate_time_exists(test_df_year):
```